rpc/common/moc_common_common.proto

```diff
@@ -112,6 +112,21 @@ message Status {
 	ValidationStatus validationStatus = 6;
 }
 
+// in the current implementation, cloudagent objects are generally identified by their group name 
+// and name. These fields are used to identify if a POST request should be treated as a create or 
+// an update
+message CloudSubResource {
+	string groupName = 1;
+	string name = 2;
+	string realizedName = 3;
```

Comment:
When we update the nodeagent availability set, the virtual machines need to be referenced by realized names so they can be found in the nodeagent store.

This just provides a way to pass down the realized name to the nodeagent avset.

We originally just wanted the cloudagent avset to just reference the name and the nodeagent avset to just reference the realized name with a translation when calling into the wssd sdk to keep the models cleaner.  We reverted to this because the avset provider cannot call back into the vm provider at wssd sdk call time to do the translation. (the vm provider references the avset provider to add members on VM creation.  The avset provider then cannot call back into the vm provider (no circular dependencies))

Reply:
Does the realized name show up when we list/print the avset cloud agent object using moccli? The realized name is different from the name in the cloud agent for security purposes. avset should show the realized name when we print the object.

Reply:
Do you mean should not print the realized name?

The idea was to filter out the realized name from the moc api view (the realized name just being internally held and utilized) and only returning the name itself.

@liwonMS for response given she is working in cloudagent

Reply:
Yes

Reply:
@raghavendra-nataraj see above for realized name